python/ray/serve/_private/utils.py

```diff
@@ -734,3 +735,13 @@ def get_all_live_placement_group_names() -> List[str]:
             live_pg_names.append(pg_name)
 
     return live_pg_names
+
+
+def in_ray_driver_process_GCS_UNSAFE() -> bool:
```

Comment:
I don't think this fails when GCS is down (would be very odd if runtime_context depends on the GCS)

Reply:
I tested it by running a Serve app on a 2-node Ray cluster with GCS fault tolerance enabled. All Serve replicas started on the worker node. The Serve replicas ran `ray.get_runtime_context()`.

I killed the head node and tried sending requests to the replicas, but the requests hung. I assume that means `ray.get_runtime_context()` doesn't work when the GCS is down.